src/datasources/asset-calibration/AssetCalibrationDataSource.test.ts

```diff
@@ -370,4 +371,61 @@ describe('queries', () => {
 
     await expect(datastore.query(buildCalibrationForecastQuery(monthBasedCalibrationForecastQueryMock))).rejects.toThrow()
   })
+
+  test('validate DAY grouping', async () => {
+    const request = buildCalibrationForecastQuery(dayBasedCalibrationForecastQueryMock);
+    const numberOfDays = 31 * 3 + 1;
+    request.range = { from: dateTime().subtract(numberOfDays, 'day'), to: dateTime(), raw: { from: `now-${numberOfDays}d`, to: 'now' } };
+    let exception: Error | undefined = undefined;
+
+    try {
+      await datastore.query(request);
```

Comment:
can't you do something like expect(() =>  datastore.query(request)).toThrow(TypeError)?

or evenBetter:

 expect(() =>  datastore.query(request)).toThrowErrorMatchingSnapshot().

Then you can change the error Snapshot to match expected message

Reply:
expect(datastore.query(request)).rejects.toThrow(
      'check for error message here'
    );

Reply:
I did go with the first approach, but I was able to specify the error message. I think it is the cleanest way